packages/components/icon/src/icon.js

```diff
@@ -39,12 +41,69 @@ export class TSIcon extends TSElement {
 			/** 90, 180, 270 */
 			rotate: { type: Number, reflect: true },
 			/** 'h', 'horizontal', 'v', 'vertical' */
-			flip: { type: String, reflect: true }
+			flip: { type: String, reflect: true },
+			/** INTERNAL svg content to be rendered inside icon */
+			svgContent: { type: String, reflect: false, attribute: null }
 		};
 	}
 
+	attributeChangedCallback(name, oldVal, newVal) {
+		super.attributeChangedCallback(name, oldVal, newVal);
+		switch (name) {
+			case 'icon':
+				if (!this.src && newVal !== null) {
+					window.customElements.whenDefined('ts-icon').then(() => {
+						this.getIcon();
+					});
+				}
+				break;
+			case 'src':
+				if (newVal !== null) {
+					window.customElements.whenDefined('ts-icon').then(() => {
+						this.getSrc();
+					});
+				}
+				break;
+			// do nothing
+		}
+	}
+
+	async getIcon() {
+		if (!this.iconCache.has(this.icon) && this.iconCache.get(this.icon) !== '') {
+			try {
+				const { default: svgUrl } = await import(`@tradeshift/elements.icon/lib/assets/icons/${this.icon}.svg`);
+				const res = await fetch(svgUrl);
+				const content = await this.extractSvgContent(res);
+				this.iconCache.set(this.icon, content);
+			} catch (e) {
+				console.error(e);
+			}
+		}
+		this.svgContent = this.iconCache.get(this.icon);
+	}
+
+	async getSrc() {
+		if (!this.iconCache.has(this.src) && this.iconCache.get(this.src) !== '') {
+			try {
+				const res = await fetch(this.src);
+				const content = await this.extractSvgContent(res);
+				this.iconCache.set(this.src, content);
+			} catch (e) {
+				console.error(e);
+			}
+		}
+		this.svgContent = this.iconCache.get(this.src);
```

Comment:
This will assign a value from the cache to this.svgContent no matter if it succeeds to import and add the icon to the cache or not.